plugins/vpp/l3plugin/vppcalls/vpp1908/route_vppcalls.go

```diff
@@ -114,6 +109,56 @@ func (h *RouteHandler) VppDelRoute(route *vpp_l3.Route) error {
 	return h.vppAddDelRoute(route, swIfIdx, true)
 }
 
+func setFibPathNhAndProto(ipStr string) (nh ip.FibPathNh, proto ip.FibPathNhProto) {
+	netIP := net.ParseIP(ipStr)
+	if netIP == nil {
+		return
+	}
+	var ipData [16]byte
+	if netIP.To4() == nil {
+		proto = ip.FIB_API_PATH_NH_PROTO_IP6
+		copy(ipData[:], netIP[:])
+	} else {
+		proto = ip.FIB_API_PATH_NH_PROTO_IP4
+		copy(ipData[:], netIP[12:])
+	}
+	return ip.FibPathNh{
+		Address: ip.AddressUnion{
+			XXX_UnionData: ipData,
+		},
+		ViaLabel:           NextHopViaLabelUnset,
+		ClassifyTableIndex: ClassifyTableIndexUnset,
+	}, proto
+}
+
+func setRoutePrefix(dstNetwork string) (ip.Prefix, error) {
+	addr, isIPv6, err := addrs.ParseIPWithPrefix(dstNetwork)
+	if err != nil {
+		return ip.Prefix{}, err
+	}
+	mask, _ := addr.Mask.Size()
+	return ip.Prefix{
+		Address: ip.Address{
```

Comment:
When you've parsed the IP address from the CIDR you can just single if statement to set both parts of `Address` field instead of doing two extra function calls, just like in `ipToAddress`: https://github.com/ligato/vpp-agent/blob/2fa8373919019b61c7796f141b66f389182cac9b/plugins/vpp/l3plugin/vppcalls/vpp1908/vppcalls_handlers.go#L187-L197